shopinvader/models/shopinvader_variant.py

```diff
@@ -50,20 +50,28 @@ class ShopinvaderVariant(models.Model):
 
     @contextmanager
     @api.multi
-    def _action_product_disabled(self):
+    def _action_product_toggle_active(self):
         """
         Action a deactivation of a variant, if every variants are disabled:
         disable the product too.
+        Also when a variant is enabled, the related shopinvader product
+        should be re-enabled too.
         :return:
         """
         product_active_dict = {
             p: p.active for p in self.mapped("shopinvader_product_id")
         }
         yield
+        shop_products_to_active = self.env["shopinvader.product"].browse()
+        shop_products_to_inactive = self.env["shopinvader.product"].browse()
         for variant in self:
+            shopinv_product = variant.shopinvader_product_id
             if variant.active:
+                # If the variant is active and the related shop. product is
+                # not active, we have to active it.
+                if not shopinv_product.active:
+                    shop_products_to_active |= shopinv_product
```

Comment:
overall looks very good, tnx.
One more thing tho: if you expect to have big amount of records here it's better to use a set of ids instead of browse records -> they are very slow if you intersect them.

You should have something like:

```py
to_activate_ids = set()
to_inactivate_ids = set()
[...]
to_activate_ids.add(shopinv_product.id)
[...]
to_inactivate_ids.add(shopinv_product.id)
[...]
self.env["shopinvader.product"].browse(list(to_activate_ids)).write({"active": True})
self.env["shopinvader.product"].browse(list(to_inactivate_ids)).write({"active": True})
```